### PR TITLE
Marker pixel info

### PR DESCRIPTION
**Pull request by JakeBrawer**
The Marker msg type now contains data relating to pixel coordinates of a marker's corners and center.

### CODE REVIEW

File: aruco_ros/launch/baxter_marker_publisher.launch [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<launch>
```
Comment (bmagyar) [inline aruco_ros/launch/baxter_marker_publisher.launch:1]: This file should not be here

File: aruco_ros/scripts/ik_service_client+mover.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
```
Comment (bmagyar) [inline aruco_ros/scripts/ik_service_client+mover.py:1]: This file should not be here

File: aruco_ros/src/marker_publish.cpp [hunk 3]
```diff
@@ -44,9 +45,18 @@ or implied, of Rafael Muñoz Salinas.
 #include <sensor_msgs/image_encodings.h>
 #include <aruco_ros/aruco_ros_utils.h>
 #include <aruco_msgs/MarkerArray.h>
+#include <geometry_msgs/Point.h>
 #include <tf/transform_listener.h>
+#include <tf/transform_broadcaster.h>
 #include <std_msgs/UInt32MultiArray.h>
 
+string int_to_string( const int a )
```
Comment (bmagyar) [inline aruco_ros/src/marker_publish.cpp:53]: Please handle this in the minimal invasive way.
This seems like a good alternative to adding a new function: http://www.boost.org/doc/libs/1_55_0/doc/html/boost_lexical_cast/examples.html

File: aruco_ros/src/marker_publish.cpp [hunk 4]
```diff
@@ -86,11 +96,13 @@ class ArucoMarkerPublisher
     , useCamInfo_(true)
   {
     image_sub_ = it_.subscribe("/image", 1, &ArucoMarkerPublisher::image_callback, this);
+    //image_sub_ = it_.subscribe("/cameras/left_hand_camera/image", 1, &ArucoMarkerPublisher::image_callback, this);
```
Comment (bmagyar) [inline aruco_ros/src/marker_publish.cpp]: Please remove this line.

File: aruco_ros/src/marker_publish.cpp [hunk 5]
```diff
@@ -86,11 +96,13 @@ class ArucoMarkerPublisher
...
 
     nh_.param<bool>("use_camera_info", useCamInfo_, true);
     if(useCamInfo_)
     {
       sensor_msgs::CameraInfoConstPtr msg = ros::topic::waitForMessage<sensor_msgs::CameraInfo>("/camera_info", nh_);//, 10.0);
+      //sensor_msgs::CameraInfoConstPtr msg = ros::topic::waitForMessage<sensor_msgs::CameraInfo>("/cameras/left_hand_camera/camera_info", nh_);//, 10.0);
```
Comment (bmagyar) [inline aruco_ros/src/marker_publish.cpp]: Please remove this line.

File: aruco_ros/src/marker_publish.cpp [hunk 6]
```diff
@@ -202,9 +225,20 @@ class ArucoMarkerPublisher
               transform = static_cast<tf::Transform>(cameraToReference) * transform;
               tf::poseTFToMsg(transform, marker_i.pose.pose);
               marker_i.header.frame_id = reference_frame_;
+
+              tf::TransformBroadcaster br;
+              br.sendTransform(tf::StampedTransform(transform, ros::Time::now(), reference_frame_,
+                                                    "/marker_"+int_to_string(marker_i.id)));
+              // printf("%s %s\n", reference_frame_.c_str(), ("/marker_"+int_to_string(marker_i.id)).c_str());
```
Comment (bmagyar) [inline aruco_ros/src/marker_publish.cpp:230]: Please remove this line.

File: aruco_ros/src/marker_publish.cpp [hunk 7]
```diff
@@ -265,6 +299,7 @@ class ArucoMarkerPublisher
 
 int main(int argc,char **argv)
 {
+  ROS_INFO("Starting ARUCO Marker Publisher");
```
Comment (bmagyar) [inline aruco_ros/src/marker_publish.cpp:294]: Please remove this line.

File: aruco_ros/src/marker_publish.cpp [hunk 8]
```diff
@@ -202,9 +225,20 @@ class ArucoMarkerPublisher
...
               marker_i.header.frame_id = reference_frame_;
+
+              tf::TransformBroadcaster br;
+              br.sendTransform(tf::StampedTransform(transform, ros::Time::now(), reference_frame_,
+                                                    "/marker_"+int_to_string(marker_i.id)));
+              // printf("%s %s\n", reference_frame_.c_str(), ("/marker_"+int_to_string(marker_i.id)).c_str());
             }
           }
 
+          if (markers_.size()>0)
```
Comment (bmagyar) [inline aruco_ros/src/marker_publish.cpp]: This is just for debugging. This information is being published. Please remove it.

**Review by bmagyar — COMMENTED** [review]
Please sort out the conflict with the package.xml as well.

File: aruco/src/aruco/marker.cpp [hunk 9]
```diff
@@ -287,8 +286,7 @@ namespace aruco {
     //rotate the X axis so that Y is perpendicular to the marker plane
     if (setYPerpendicular) rotateXAxis(Rvec);
     ssize=markerSizeMeters;
-    cout<<(*this)<<endl;
-    
+    cout<<"Marker "<<(*this)<<endl;
```
Comment (bmagyar) [inline aruco/src/aruco/marker.cpp]: This needs to be removed

File: aruco_ros/CMakeLists.txt [hunk 10]
```diff
@@ -13,34 +13,36 @@ find_package(catkin REQUIRED COMPONENTS
...
 include_directories(
   include
   ${catkin_INCLUDE_DIRS}
 )
 
-add_executable(single src/simple_single.cpp
-                      src/aruco_ros_utils.cpp)
-add_dependencies(single ${PROJECT_NAME}_gencfg)
-target_link_libraries(single ${catkin_LIBRARIES})
+add_library(aruco_ros
```
Comment (bmagyar) [inline aruco_ros/CMakeLists.txt:33]: The library name could be misleading. I think aruco_ros_utils would be better

**Review by bmagyar — CHANGES_REQUESTED** [review]
I've added a few new ones but most of the old requested changes still remain.

**Comment by jdlangs** [comment]
There doesn't seem to be any way to deduce the added message fields are in pixels without reading the code. Ideally, it should be obvious from the names of the fields themselves.

**Comment by jdlangs** [comment]
Also, am I just missing something or does the new `center` message field never get populated?

File: aruco_ros/src/marker_publish.cpp (hunk 3 above)
Comment (jdlangs) [reply aruco_ros/src/marker_publish.cpp:53]: If it's ok to add the C++11 build flag to `CMakeLists.txt`, `std::to_string` would be the way to go.

**Comment by bmagyar** [comment]
Thanks @jdlangs for the extra notes, the center could be removed as it's trivial to compute it from the corners.

File: aruco_ros/src/marker_publish.cpp (hunk 3 above)
Comment (bmagyar) [reply aruco_ros/src/marker_publish.cpp:53]: :+1:

**Comment by JakeBrawer** [comment]
Hello all, thank you for your timely and trenchant comments. I actually meant to open this pull request on our own fork of this repo, but obviously failed to do so as I am a noob apparently :^).  Do you think you can terminate this PR (or tell me how to do it!). Unless you find any of these changes useful, in which case I'll make the changes.

Thanks again, and sorry for the confusion.

**Pull request closed without merging**